blockchain/validator_set.go

```diff
@@ -0,0 +1,273 @@
+package blockchain
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math"
+	"sort"
+	"strconv"
+
+	acm "github.com/hyperledger/burrow/account"
+	rpc_core "github.com/tendermint/tendermint/rpc/core"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	tm_types "github.com/tendermint/tendermint/types"
+)
+
+type ValidatorSet interface {
+	TotalPower() int
+	SetMaximumPower(maximumPower int)
+	MaximumPower() int
+	AdjustPower(height int64) error
+	Validators() []acm.Validator
+	SetLeavers() []acm.Validator
+	JoinToTheSet(validator acm.Validator) error
+	LeaveFromTheSet(validator acm.Validator) error
+	IsValidatorInSet(address acm.Address) bool
```

Comment:
I suggest `Join`, `Leave`, and `Contains` are better more idiomatic names

Reply:
Thanks